community/kernel/src/main/java/org/neo4j/kernel/impl/api/KernelTransactionImplementation.java

```diff
@@ -131,7 +131,7 @@ TransactionType upgradeToSchemaTransaction() throws InvalidTransactionTypeKernel
     private TransactionHooks.TransactionHooksState hooksState;
     private KernelStatement currentStatement;
```

Comment:
From your description i can conclude that TransactionType represent currently what kind of writes where performed in transaction. In this case it sounds reasonable to me to call initial state NONE, rename TransactionState to something that will actually name correctly what was just described and it should be javadoced in the code